### DISCUSSION TITLE
GH Packages Pricing and Install with GITHUB_TOKEN

### DISCUSSION THREAD

**Discussion by canastro** [community/community#25140] [Actions]
<p>From what I’ve read in some other questions is that its not possible to use GITHUB_TOKEN to install packages from other repos, and you should instead use a personal token.</p>
<p>In the pricing table on GH packages it says: “Data transfer out within Actions” = “Unlimited”  and in the about page of github packages says: “When you create a GitHub Actions workflow, you can use the GITHUB_TOKEN to publish and <strong>install packages</strong> in GitHub Packages without needing to store and manage a personal access token.”</p>
<p>My question is, how is github going to price correctly the installs if we’re not able to use the GITHUB_TOKEN to install our GH packages from GH Actions?</p>
<p>If this doesn’t work, then that pricing table feels a bit like false advertisement.  </p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/canastro">@canastro</a> ,</p>
<p>If the package is on the other repo, GITHUB_TOKEN doesn’t have enough access to pull the package and used to install it, then personal access token is needed. This will require payment.</p>
<p>If you directly push and pull your packages from within your workflow, GITHUB_TOKEN is permission enough, it should be unlimited. </p>
<p>You can refer to similar ticket <a href="https://github.community/t/getting-charged-for-data-transfer-from-packages-to-actions/17006/2">here</a> for more details.</p>
<p>Thanks.</p>

**Comment by canastro** [comment]
<p>But I don’t think it makes sense to pay for installs on packages within the same organization. Thats not clear in the pricing and anywhere in the documentation.</p>
<p>How does GH Packages identifies if a install came from GH Actions? Is by checking the token used? </p>
<p>The only way that we can use GITHUB_TOKEN to install from GH Packages is with a mono-repo approach?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/canastro">@canastro</a> ,</p>
<p>The only way that we can use GITHUB_TOKEN to install from GH Packages is with a mono-repo approach?</p>
<p>&gt;&gt; it’s accurate.</p>
<p>How does GH Packages identifies if a install came from GH Actions? Is by checking the token used? </p>
<p>&gt;&gt;It’s based on ip-address, hosted VM’s have known ip range.</p>
<p>More details please check official link:</p>
<p><a href="https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-packages" rel="nofollow noopener">https://help.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-packages</a></p>
<p>Thanks.</p>
